src/main/kotlin/com/fortysevendeg/arrowinpractice/endpoints/CharacterDetails.kt

```diff
@@ -1,15 +1,49 @@
 package com.fortysevendeg.arrowinpractice.endpoints
 
+import arrow.Kind
+import arrow.core.Option
+import arrow.core.toOption
+import arrow.effects.*
+import arrow.typeclasses.binding
 import com.fortysevendeg.arrowinpractice.database.CharactersDatabase
 import com.fortysevendeg.arrowinpractice.error.InvalidIdException
 import com.fortysevendeg.arrowinpractice.error.NotFoundException
-import com.fortysevendeg.arrowinpractice.serialization.characterId
+import com.fortysevendeg.arrowinpractice.model.Character
+import io.ktor.application.ApplicationCall
 import io.ktor.application.call
 import io.ktor.auth.authenticate
-import io.ktor.request.path
 import io.ktor.response.respond
+import io.ktor.routing.Route
 import io.ktor.routing.Routing
 import io.ktor.routing.get
+import io.ktor.util.pipeline.ContextDsl
+import io.ktor.util.pipeline.PipelineContext
+
+private fun paramOf(name: String, app: PipelineContext<Unit, ApplicationCall>): Option<String> =
+  app.call.parameters[name].toOption()
+
+private fun IO.Companion.characterIdOrNotFound(maybeId: Option<String>): IO<String> =
```

Comment:
maybe `getOrNotFound(maybeId: Option<String>)` for simplicity ? 

also: is this an IO because it throws? just curious. feels like attendees could get a bit confused as I am now, since this seems to force unwrap the inner value or throw. In case it succeeds we lift it to IO, and I suspect that's just because we wanna make it part of the binding syntax, but we end up calling `bind()` right away after calling this function, so, what's the benefit of wrapping into IO and unwrapping right away ?

Maybe we could do this but with a more justified blocking op like taking something else from another table in the DB and then combining? something that makes more sense on a real implementation.